Send router advertisements

drbild opened the pull request:
Work in progress implementation of #48 . 

All that is missing is implementing the `ip_prefix` function in `ip.c` to parse IPv6 addresses with prefix lengths into the constituent address and prefix length fields.

peterhaijen commented on src/icmp.c:
```diff
@@ -0,0 +1,166 @@
+    if (i == 1)
+        sum += (uint16_t) *(uint8_t*)w;
+
+    sum += sum >> 16;
```
I think this is wrong. Should be `while (sum>>16) sum = (sum>>16) + (sum&0xFFFF);` Consider what happens when sum = 0x2FFFF:

- When sum = 0x2FFFF, correct value for sum (before 1s complement) would be 0xFFFF + 0x2 = 0x10001, then 0x0001 + 0x1 = 0x0002. Checksum becomes ~0x0002 = 0xfffd.

With this code, sum becomes 0x30001. Checksum then becomes 0xfffe.

drbild replied:
Yes, thanks! I forgot to update this function based on your xbridge branch. Will fix.

drbild commented:
@peterhaijen 

I added your prefix parsing code (the easier-on-the-eyes) variant.  Thanks for doing that! I made a couple changes to improve error handling (moved from `atoi` to `strtol).

I also fixed the checksum computation.

After you've taken one more look, I'll merge this.

Linux by default ignores route options with a prefix longer than 0 (i.e., for anything but the default route). That can be changed via a `sysctl` param, but requires config on the host.  For the router card we'll only be sending the default route, so it's ok as is.

`enftun-setup` still sets the routes manually via `ip -6 route`, for non-router-card users.

`enftun-setup` does a bunch of that sort of routing and firewall setup that we don't want for `enf0` on the router card.   I'll open a separate PR to make those configuration next week.  For now, we can just use a custom systemd script for enf0 that doesn't run enftun-setup.
